workflows/ctag_Wc_valid_sf.py

```diff
@@ -0,0 +1,483 @@
+            elif 'sl_' in histname:
+                fields = {l: ssmu[l] for l in h.fields if l in dir(ssmu)}
+                if(isRealData): h.fill(dataset=dataset,char=osss,**fields,weight=osss)
+                else: h.fill(dataset=dataset,char=osss,**fields,weight=weights.weight()[event_level]*osss)
+            elif 'met' in histname: 
+                fields = {l: selev.METFixEE2017[l] for l in h.fields if l in dir(selev.METFixEE2017)}
+                if(isRealData): h.fill(dataset=dataset,**fields,weight=osss)
+                else: h.fill(dataset=dataset,**fields,weight=weights.weight()[event_level]*osss)
+            # elif 'mujet_' in histname:
+            #     fields = {l: smuon_jet[l] for l in h.fields if l in dir(smuon_jet)}
+            #     if isRealData: h.fill(dataset=dataset,flav=5, char=osss, **fields,weight=osss)
+            #     else :h.fill(dataset=dataset,flav=5, char=osss, **fields,weight = weights.weight()[event_level]*osss)
+        
+        if not isRealData:
+            ###Fill no SFs
+            output['zmass'].fill(dataset=dataset,char=osss,zmass=sz.mass,weight=weights.weight()[event_level]*osss)
+            output['zpt'].fill(dataset=dataset,char=osss,zpt=sz.pt,weight=weights.weight()[event_level]*osss)
+            output['zeta'].fill(dataset=dataset,char=osss,zeta=sz.eta,weight=weights.weight()[event_level]*osss)
+            output['zphi'].fill(dataset=dataset,char=osss,zphi=sz.phi,weight=weights.weight()[event_level]*osss)
+            output['wmass'].fill(dataset=dataset,char=osss,wmass=sw.mass,weight=weights.weight()[event_level]*osss)
+            output['wpt'].fill(dataset=dataset,char=osss,wpt=sw.pt,weight=weights.weight()[event_level]*osss)
+            output['weta'].fill(dataset=dataset,char=osss,weta=sw.eta,weight=weights.weight()[event_level]*osss)
+            output['wphi'].fill(dataset=dataset,char=osss,wphi=sw.phi,weight=weights.weight()[event_level]*osss)
+            output['drmumu'].fill(dataset=dataset,char=osss,drmumu=ssmu.delta_r(shmu),weight=weights.weight()[event_level]*osss)
+            output['hlptratio'].fill(dataset=dataset,flav=genflavor,char=osss,hlptratio=shmu.pt/smuon_jet.pt,weight=weights.weight()[event_level]*osss)
+            output['slptratio'].fill(dataset=dataset,flav=genflavor,char=osss,slptratio=ssmu.pt/smuon_jet.pt,weight=weights.weight()[event_level]*osss)
+            output['drjet_hmu'].fill(dataset=dataset,flav=genflavor,char=osss,drjet_hmu=smuon_jet.delta_r(shmu),weight=weights.weight()[event_level]*osss)
+            output['drjet_smu'].fill(dataset=dataset,flav=genflavor,char=osss,drjet_smu=smuon_jet.delta_r(ssmu),weight=weights.weight()[event_level]*osss)
+
+            ## discri
+            
+            output['btagDeepFlavB_0'].fill(dataset=dataset,flav=genflavor,char=osss, btagDeepFlavB=smuon_jet.btagDeepFlavB,weight=weights.weight()[event_level]*osss)
```

Comment:
Could these be filled programmatically like in the example wf https://github.com/cms-btv-pog/BTVNanoCommissioning/blob/master/workflows/ttbar_validation.py#L188

Reply:
Fair enough, it's probably not worth rewriting, but I think the implementation of the SF weights could have been simplified by having them included as a variation in the `weights` object